lib/active_model/serializer/adapter/attributes.rb

```diff
@@ -51,10 +23,76 @@ def fragment_cache(cached_hash, non_cached_hash)
 
         private
 
+        def serializable_hash_for_collection(options = {})
+          serializer.map { |s| Attributes.new(s, instance_options).serializable_hash(options) }
+        end
+
+        def serializable_hash_for_single_resource(options = {})
+          resource = resource_object_for(serializer, options)
+          resource.merge!(resource_relationships)
+        end
+
+        # iterate through the associations on the serializer,
+        # adding them to the parent as needed (as singular or plural)
+        #
+        # nested_associations is a list of symbols that governs what
+        # associations on the passed in seralizer to include
+        def resource_relationships
+          relationships = {}
+
+          serializer.associations(@include_tree).each do |association|
+            association_serializer = association.serializer
+            opts = association.options
+            key = association.key
+            opts[:include] = @include_tree[key].to_hash
+
+            if association_serializer.respond_to?(:each)
+              relationships[key] = to_many_relationship_for(association_serializer, opts)
+            else
+              relationships[key] = to_one_relationship_for(association_serializer, opts)
+            end
+          end
+
+          relationships
+        end
+
+        # add a singular relationship
+        # the options should always belong to the serializer
+        def to_one_relationship_for(serializer, options)
+          if options[:include].present?
+            Attributes.new(serializer, options).serializable_hash(options)
```

Comment:
That is because you should do `instance_options.merge(include: include_tree[association.key])`, and possibly add a case to `IncludeTree.from_include_args` to handle an IncludeTree directly (as in #1157).